src/goto-symex/goto_symex_state.cpp

```diff
@@ -688,6 +688,8 @@ void goto_symex_statet::rename(
   const namespacet &ns,
   levelt level)
 {
+  type = ns.follow(type);
```

Comment:
I'm not even sure that compiles, but even when you make it compile: I think it should be done after `requires_renaming` to make sure we don't unnecessarily break sharing.